diktat-rules/src/test/resources/test/paragraph3/string_concatenation/StringConcatenationTest.kt

```diff
@@ -0,0 +1,24 @@
+package test.chapter3.strings
+
+val valueStr = "my str"
+
+val myTest1 = "my string " + "string " + valueStr + " other value"
+val myTest2 = "my string " + 1 + valueStr + " other value"
+val myTest3 = "my string " + valueStr + " string " + "other value"
+val myTest4 = "my string" + (" one " + "two")
+val myTest5 = "trying to sum with " + """multiline"""
+val myTest6 = "trying to sum with " + """
+    multiline""".trimIndent()
+val myTest7 = """multiline""" + " string"
+val myTest8 = "string " + valueStr.replace("my", "")
+val myTest9 = "string " + "valueStr".replace("my", "")
+val myTest10 = "string " + (1 + 5)
+val myTest11 = "sum " + ("other string " + 3 + " str2 " + 5 + (" other string 2 " + 2 + " str3 " + 4))
+val myTest12 = "my string " + 1 + 2 + 3
+val myTest13 = (1 + 2).toString() + " my string " + 3 + " string " + valueStr + valueStr
```

Comment:
need to add the following test:
`a.toString() + "fghgfhfg"` should not be fixed to "$afghgfhfg", but should be fixed to "${a}fghgfhfg"

`a.toString() + " fghgfhfg"` should be fixed to "$a fghgfhfg"

Reply:
We can make it even easier: in string concatenation rule always put braces, and then SRING_TEMPLATE_CURLY_BRACES inspection will remove redundant ones